platform/ios/PolyPodApp/PolyPod/Authentication.swift

```diff
@@ -35,21 +35,48 @@ class Authentication {
         UserDefaults.standard.set(true, forKey: Authentication.disableCheckKey)
     }
     
-    func setUp(_ completeAction: @escaping (Bool) -> Void) {
-        authenticateLocally(reason: "auth_prompt_set_up") { success in
+    func setUp(_ reAuth: Bool, _ completeAction: @escaping (Bool) -> Void) {
+        
+        let reason = reAuth ?
+        "re_auth_prompt_set_up": "auth_prompt_set_up"
+        
+        authenticateLocally(reason) { success in
             self.authenticated = success
+
+            if (!success) {
+                completeAction(false)
+                return
+            }
+            
             UserDefaults.standard.set(true, forKey: Authentication.setUpKey)
             completeAction(success)
         }
     }
     
-    func authenticate(_ completeAction: @escaping (Bool) -> Void) {
+    func disable(_ completeAction: @escaping (Bool) -> Void) {
+        authenticateLocally("auth_prompt_disable") { success in
+            self.authenticated = false
```

Comment:
This seems similar to lines 44-53 above, except for the first line. Can it be factored out to a function, in the spirit of DRY?

Reply:
mmh I wouldn't say so, the setting to the Authentication.setUpKey has different value, as well as the title of the "reason". With one function we would end up in ugly if-elses :/

Reply:
46 to 49 are the same as 59 to 62 below. Differences in 58-44 (`success` or `false`) and 64-51 (`true` or `false`). I don't know if this is contingent on the other, if it's, simply passing one argument would be enough, if it's not, two arguments, one for "authenticated" and another for `UserDefaults.standard.set`

Not entirely clear to me why `success` is used as an alternative to `false`. Is there any added metadata in `success`?

Reply:
well it's different logic - on the `disable()` we set the `authenticated` to false, on the `authenticate()` the success (true) is ok.
I feel if we make one function it will be more complex on reading it - with if-elses for the `reason` and for different value setting. Regardless if we pass a second arg for the `newValue`.